### ISSUE TITLE
Update to one field cleared ALL EA values.

### ISSUE THREAD

**Issue by limonesJM** [mike-levenick/mut#36]
I added asset tags to all my machines and the upload worked great. Unfortunately, all values for my extension attributes were removed at the same time.  I did not have columns for the EAs in the template and am wondering if that was the cause.

**Comment by mike-levenick** [comment]
It certainly should not have.

- What version of Jamf Pro?
- What version of MUT?
- Can you send me your CSV file please?
- Are the EAs populated by scripts or are they dropdown or just text boxes?

**Comment by mike-levenick** [comment]
Update: I just re-created this. I'm investigating now.

**Comment by limonesJM** [comment]
Wonderful.  I was just in the process of providing the requested information:
Jamf Pro 10.14
MUT 5.0.3
Can I send you my CSV with different entries? (I can't send out actual serial no. and asset tags)
EAs are populated by scripts

THANK YOU

**Comment by mike-levenick** [comment]
Hello,

I am really sorry about this. I discovered the issue. Basically, when MUT builds the XML, it includes all the subgroups (purchasing, general, location, etc.) regardless of whether or not they are populated.

It seems that the Jamf Pro API treats the "extension_attributes" subgroup differently than any of the others, and if it is included with nothing in it, it nukes all EAs, instead of ignoring it.

With all other subgroups, it simply ignores those subgroups if they are empty.

I have corrected the issue in MUT and am building a fixed version 5.0.5 presently.
